front/lib/api/data_sources.ts

```diff
@@ -416,9 +418,13 @@ export async function upsertTable({
     tableParents.push(nonNullTableId);
   }
 
+  const useAppForHeaderDetectionFlag = auth
+    .getNonNullableWorkspace()
+    .flags.includes("use_app_for_header_detection");
+
   if (async) {
     // Ensure the CSV is valid before enqueuing the upsert.
-    const csvRowsRes = csv ? await rowsFromCsv(csv) : null;
+    const csvRowsRes = csv ? await rowsFromCsv(auth, csv, false) : null;
```

Comment:
I'm also wondering if we should pass the proper value instead of "false" while validating the table. ( as mentioned here : https://github.com/dust-tt/dust/pull/7077#discussion_r1761117274 ). It will make the calls be done twice, but if it's cached it may be ok ?